app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_people_list = []
+    for person_info in people:
+        name = person_info["name"]
+        age = person_info["age"]
+        person = Person(name, age)
+        new_people_list.append(person)
+
+    for person_info in people:
+        name = person_info["name"]
+        person = Person.people[name]
+
+        if "wife" in person_info and person_info["wife"]:
```

Comment:
Use the `dict.get()` method instead of checking for the presence of a key with `in` and accessing it directly. This approach simplifies the code and avoids potential `KeyError` exceptions.